src/libasr/pass/print_arr.cpp

```diff
@@ -100,15 +111,38 @@ class PrintArrVisitor : public PassUtils::PassVisitor<PrintArrVisitor>
                     for (size_t j=0; j<print_body.size(); j++) {
                         body.push_back(al, print_body[j]);
                     }
-                    print_stmt = ASRUtils::STMT(ASR::make_Print_t(
-                        al, x.base.base.loc, nullptr, body.p, body.size(),
-                        nullptr, nullptr));
+                    if (x.m_separator) {
+                        print_stmt = ASRUtils::STMT(ASR::make_Print_t(
+                            al, x.base.base.loc, nullptr, body.p, body.size(),
+                            x.m_separator, x.m_separator));
+                    } else {
+                        print_stmt = ASRUtils::STMT(ASR::make_Print_t(
+                            al, x.base.base.loc, nullptr, body.p, body.size(),
+                            nullptr, space));
+                    }
                     pass_result.push_back(al, print_stmt);
-                    pass_result.push_back(al, empty_print_endl);
                     print_body.clear();
                 }
                 print_stmt = print_array_using_doloop(x.m_values[i], x.base.base.loc);
                 pass_result.push_back(al, print_stmt);
+                pass_result.push_back(al, back);
```

Comment:
Also, I was a little worried about this backspace!

Reply:
yeah . I'll give a brief of why I had to use a `backsapce`. So on main , if we print an array we get this 
```
(lf) anutosh491@spbhat68:~/lpython/lpython$ cat examples/expr2.py 
def main0():
    x: i32 = 2
    arr1: i32[x]
    arr1[0] = 5
    arr1[1] = 10
    print(arr1)

main0()
(lf) anutosh491@spbhat68:~/lpython/lpython$ ./src/bin/lpython examples/expr2.py 
5
10

```
So each element is being printed on a newline and then there's an empty line .
Now first I've fixed this through my changes in the `print_array_using_doloop` function. I've added the following 
```
+        ASR::ttype_t *str_type_len_1 = ASRUtils::TYPE(ASR::make_Character_t(
+       al, loc, 1, 1, nullptr, nullptr, 0));
+        ASR::expr_t *space = ASRUtils::EXPR(ASR::make_StringConstant_t(
+       al, loc, s2c(al, " "), str_type_len_1));
....
                ASR::stmt_t* print_stmt = ASRUtils::STMT(ASR::make_Print_t(al, loc, nullptr,
-                                                                print_args.p, print_args.size(), nullptr, nullptr));
-                                                                print_args.p, print_args.size(), nullptr, space));
```
This function basically supports printing of the elements of the array . And the following change will change the structure from something like 
```
5
10

```
to 
```
5 10 # where 10 is followed by a space 
```
Hence I introduced a `backspace` character there to get rid of the space after 10